### ISSUE TITLE
EIA 2.4.1 - Audit Enablement causing Expand issue in Analyst Notebook

### ISSUE THREAD

**Issue by aq94** [i2group/analyze#67]
Hi Team,

We've been facing some challenges after implementing the database audit logging feature following the steps outlined in the links below:

- https://github.com/i2group/analyze/blob/master/documentation/java-api/Preparing-to-use-i2-Analyze-Developer-Essentials.md
- https://github.com/i2group/analyze/blob/master/documentation/java-api/Configuring-the-audit-logging-to-database-example-project.md

To provide some context, in our previous version of EIA (2.3.0) we had followed the same steps and faced no issues. Earlier this year in May we upgraded to EIA 2.4.1 which is where we began to face the expand issue - after following the steps in the links above and redeploying to enable auditing, the expand operation in Analyst Notebook does not work and gives a couple of pop-up errors messages (see attached).

We raised a case on this matter with i2 Support (Case: 00007639) where we were able to establish that the audit logging implementation was causing the expand error in ANB, so we disabled the audit logging to allow the expand to function. As part of these discussions we thought it was due to us using a jar file compiled using the older version of EIA (2.3.0) which was not compatible with our current version of EIA (2.4.1)

We are currently on i2 Analyze 4.3.5, and so I've downloaded the Developer Essentials repository v4.3.5 supplied in the i2 Github. I've followed the steps in the links and redeployed however we are still facing the expand error, which may suggest there may be another underlying issue not related to jar file compatibility.

Note: There is no loss of expand functionality in WebClient, and logs for the other actions (quick_search, visual query, etc.) are being captured in our audit tables. Expands performed in WebClient are not being captured in our audit table.

I deployed the audit logging just before 10pm last night (29/09/22) and between 10pm - 10.15pm performed some front-end tests. I've attached the console.log for reference.

Please let me know if there are any other details you need that may assist in troubleshooting.

Thanks,
Adam

[console.zip](https://github.com/i2group/analyze/files/9680707/console.zip)

<img width="316" alt="ANB Expand Error Message 1" src="https://user-images.githubusercontent.com/92896597/193189700-a41c43f0-d903-4a18-908d-a7c340911af3.PNG">
<img width="205" alt="ANB Expand Error Message 2" src="https://user-images.githubusercontent.com/92896597/193189710-5d40385a-d281-4f54-9877-f7afecbbd157.PNG">

**Comment by aq94** [comment]
Hi Anthony,

Thanks for the information above, the good news is that we've been able to successfully troubleshoot the issue.

We decided to drop our previous tables and re-create them with the new script provided in the Developer Essentials repository. Noting the two tables and columns (expand.seeds and record_retrieval.records) they were assigned the CLOB(2147483647) data type.

When we re-started ANB we were unable to perform a quick search however. This was an issue one of our other engineers faced as per the link below, so we ended up recreating the tables again and assigning CLOB(2147483647) to the USER_SECURITY_GROUPS and USER_SECURITY_PERMISSIONS columns for all of our tables.

https://github.com/i2group/analyze/issues/59

After redeploying with these changes, we haven't seen any issues in the front end and all the tables seem to be capturing the logs as expected.

Happy to close this issue as resolved and thanks again for the help.

Cheers,
Adam
